File: Arrays and Strings/Questions/1.5 One_Away.js
// 1.5 One Away
// Given two strings, write a function to chek if they are one edit or zero edits away

var oneAway = function(string1, string2) {
  // insert a char for str1 -> remove a char for str2
  var checkOneMissing = function(first, second) {
    if (first.length !== second.length - 1) {
      return false;
    } else {
      var mulligan = false;
      var fP = 0; // first Pointer
      var sP = 0; // second Pointer
      while (fP < first.length) {
        if (first[fP] !== second[sP]) {
          if (mulligan) {
            return false;
          } else {
            mulligan = true;
            sP++; // second length is longer
          }
        } else {
          fP++;
          sP++;
        }
      }
      return true;
    }
  };

File: Arrays and Strings/Questions/1.5 One_Away.rb
# One Away
# Write a function to check if a string is one edit or zero edits away

def one_edit_away(first, second)
  if (first.length == second.length)
    one_edit_replace(first, second)
  elsif (first.length + 1 == second.length)
    one_edit_insert(first, second)
  elsif (first.length - 1 == second.length)
    one_edit_insert(first, second)
  else
    return false
  end
end
